benchmarks/operator_benchmark/ops/th_intraop_test.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from operator_benchmark import benchmark_core, benchmark_runner
+from operator_benchmark.benchmark_test_generator import *
+
+import torch
+
+
+"""Microbenchmarks for TH and ATen (PyTorch) CPU intra-op parallelism.
+
+Tests the following functions:
+
+- bitor, cbitor
+    - tensor-scalar and tensor-tensor element-wise function), integer-only
+    - explicit OMP pragma, THTensorApply macros
+
+- tahn and sigmoid
+    - unary ops, THTensorMoreMath macro and UnaryOps
+
+- sumall
+    - basic reduction function
+    - pragma OMP 'parallel for' with reduction
+"""
+
+# Config
+config = generate_configs(
+    N=[128, 1024, 4096],
+    M=[128, 1024, 4096],
+    dtype=[torch.float32, torch.int32],
+    contig=[True, False],
+    mode=['short'],
+    sample_func=cross_product
+)
+
+
+def torch_or(tensor_arg):
+    def _ior(x, s):
```

Comment:
it might be more readable just with string stitching:

```
a=torch.jit.ScriptModule()
a.define("def forward(self,x):\n return x+1\n")
a(torch.tensor([5])) # works
```

you can just substitute the code instead of tracing/scripting